src/BlockTypes/ProductButton.php

```diff
@@ -168,20 +164,27 @@ protected function render( $attributes, $content, $block ) {
 				$this->prevent_cache();
 			}
 
-			$div_directives = 'data-wc-context=\'' . wp_json_encode( $context, JSON_NUMERIC_CHECK ) . '\'';
+			$interactive = array(
+				'namespace' => 'woocommerce/product-button',
```

Comment:
NIT: We could use using the variables to compose the namespace:

```suggestion
				'namespace' => $this->namespace . "/" . $this->block_name,
```